packages/ocapn/src/client/util.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+
+/**
+ * @typedef {import('../codecs/components.js').OcapnLocation} OcapnLocation
+ * @typedef {import('./types.js').LocationId} LocationId
+ */
+
+import { Buffer } from 'buffer';
+
+/**
+ * @param {Uint8Array} value
+ * @returns {string}
+ */
+export const toHex = value => {
+  return Buffer.from(value).toString('hex');
+};
+
+/**
+ * @param {OcapnLocation} location
+ * @returns {LocationId}
+ */
+export const locationToLocationId = location => {
+  return `${location.transport}:${location.address}`;
+};
+
+const swissnumDecoder = new TextDecoder('ascii', { fatal: true });
```

Comment:
I don’t believe that this usage is portable. Consider trivially hand-rolling this routine. I think it’s just something like `String.fromCode(...array)` as long as you assert that the values of the array are covered by the `0x7f` mask.